Repositori.Core/Repositories/IDeleteRepository.cs

```diff
@@ -42,6 +42,6 @@ public interface IDeleteRepository<TEntity>
         /// <returns>An awaitable task providing a list of created data objects</returns>
         Task<List<TEntity>> DeleteAsync(ICollection<TEntity> entities);
 
-        Task<ICollection<TEntity>> DeleteByAsync(Expression<Func<TEntity, bool>> filter);
+        Task<TEntity> DeleteByAsync(Expression<Func<TEntity, bool>> filter);

```

Comment:
This may have multiple deleted entities, I think this should be a collection, unless the contract explicitly states that there is only one that will be deleted

Reply:
I don't think that would be consistent with the interface for GetBy.
It's also misaligned with DeleteBy further up.

Reply:
I feel then there isn't much difference between Delete(entity) and DeleteBy in that case

Reply:
Have changed it back to returning a list, and changed the synchronouse DeleteBy to also return a list